theories/core/Core.ec

```diff
@@ -263,3 +263,8 @@ proof. by move=> x @/predI [] _ ->. qed.
 
 lemma nosmt predTofV (f : 'a -> 'b): predT \o f = predT.
 proof. by apply/fun_ext=> x. qed.
+
+lemma nosmt predI_idempotent (p : 'a -> bool):
```

Comment:
Maybe call this `predIpp` in analogy to`andbb` (or `<foo>xx`) for generic arguments, or maybe `predI_id`. Also maybe add `predU_id`?

Reply:
Fixed